src/pages/Models/index.tsx

```diff
@@ -1,113 +1,874 @@
 import CodeEditor from "@/components/CodeEditor";
+import ErrorFound from "@/components/ErrorFound";
+import ModelsSidebar from "@/components/ModelsSidebar";
+import SidebarLayout from "@/layouts/SidebarLayout";
+import Console from "@/components/Console";
+import Modal from "@/components/Modal";
+import DataModelGeneration from "@/components/DataModelGeneration";
+import VersionsList from "@/components/VersionsList";
+import useUserData from "@/hooks/useUserData";
+import useAppSettings from "@/hooks/useAppSettings";
+import useLocation from "@/hooks/useLocation";
+import useSchemasIde from "@/hooks/useSchemasIde";
+import useSchemas from "@/hooks/useSchemas";
+import useSources from "@/hooks/useSources";
+import useCheckResponse from "@/hooks/useCheckResponse";
+import usePermissions from "@/hooks/usePermissions";
+import equals from "@/utils/helpers/equals";
+import calcChecksum from "@/utils/helpers/dataschemasChecksum";
+import type { Branch, DataSourceInfo, Schema } from "@/types/dataSource";
+import type { Dataschema } from "@/types/dataschema";
 import type { Version } from "@/types/version";
-import type { File } from "@/types/file";
+import type { Branches_Insert_Input } from "@/graphql/generated";
 
 import ModelsActiveIcon from "@/assets/models-active.svg";
 
 import styles from "./index.module.less";
 
+import type { MenuProps } from "antd";
+import type { ChangeEvent } from "react";
+
 interface ModelsProps {
-  versions: Version[];
-  currentVersion: Version;
-  branches: string[];
-  docs: string;
+  dataSources: DataSourceInfo[];
+  branchMenu: MenuProps["items"];
+  ideMenu: MenuProps["items"];
+  branches: Branch[];
+  onSetDefault: (branchId?: string) => void;
+  onChangeBranch: (branchId?: string) => void;
+  onCreateBranch: (name: string) => Promise<void>;
+  onSchemaDelete: (id: string) => void;
+  onSchemaUpdate: (editId: string, values: Partial<Dataschema>) => void;
+  dataSource?: DataSourceInfo;
+  versions?: Version[];
+  currentBranch?: Branch;
+  currentVersion?: Version;
+  dataschemas?: Dataschema[];
+  onSchemaCreate: (values: Partial<Dataschema>) => void;
+  onCodeSave: (id: string, code: string) => void;
+  onRunSQL: (query: string, limit: number) => void;
+  dataSchemaName: string;
+  fetching?: boolean;
+  genSchemaModalVisible?: boolean;
+  versionsModalVisible?: boolean;
+  tablesSchema: Schema;
+  schemaFetching?: boolean;
+  onModalClose: () => void;
+  data?: object[];
+  validationError?: string;
+  isConsoleOpen?: boolean;
+  toggleConsole?: () => void;
+  onSaveVersion: (
+    checksum: string,
+    data: ({
+      user_id?: string;
+    } & Partial<Dataschema>)[]
+  ) => void;
+  onGenSubmit: (values: object, format: string) => void;
+  onDataSourceChange: (dataSource: DataSourceInfo | null) => void;
+  sqlError?: object;
 }
 
 const { Title } = Typography;
 
-const Models: React.FC<ModelsProps> = ({
-  // versions,
+export const Models: React.FC<ModelsProps> = ({
```

Comment:
file is too long, we need a refactoring here. All scripts should not be greater than 500 lines